packages/cloudflare/src/workflows.ts

```diff
@@ -98,38 +76,40 @@ class WrappedWorkflowStep implements WorkflowStep {
     configOrCallback: WorkflowStepConfig | (() => Promise<T>),
     maybeCallback?: () => Promise<T>,
   ): Promise<T> {
+    // Capture the current scope, so parent span (e.g., a startSpan surrounding step.do) is preserved
+    const scopeForStep = getCurrentScope();
+
     const userCallback = (maybeCallback || configOrCallback) as () => Promise<T>;
     const config = typeof configOrCallback === 'function' ? undefined : configOrCallback;
 
     const instrumentedCallback: () => Promise<T> = async () => {
-      return workflowStepWithSentry(this._instanceId, this._options, async () => {
```

Comment:
can it be a problem that we no longer create a sentry instance here? 🤔 is there always an existing sentry instance outside here...?